include/onnxruntime/core/common/inlined_containers_fwd.h

```diff
@@ -23,6 +23,7 @@
 #endif  // _MSC_VER
 
 #include <absl/container/inlined_vector.h>
+#include <absl/container/flat_hash_map.h>
```

Comment:
The problem with this, at least in the past, not everything compiles under CUDA. That is why a FWD header was invented to be used in headers as a forward declaration.

FWD headers minimize inclusions. I understand that this is needed for default arguments, but it leads to huge problems down the road.

Reply:
Is there a specific alternative you recommend? (Sorry, I don't know the context enough to understand, maybe this is a naive question.)

Reply:
I have not had time to think about it for longer. Ed's suggestion looks good as far as the templates go. However, if the next version of absl header fails to compile some place, we are screwed.

Reply:
I can see CUDA build is failing due to ABSL code.

Reply:
My specific suggestion is to use `std` container for the kernel and explain the kernel why and revert changes in the common container's headers. We would need to seek a long-term solution.

Reply:
The abseil flat_hash_map uses noticeably less memory with large LabelEncoders so I would like to retain it where possible. I opted to revert the container header changes and simply localise the definition of the hashmap in "labelencoder.h" where we already had an `#ifndef DISABLE_ABSEIL`.